src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -3,16 +3,26 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Remote;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal abstract class AbstractTodoCommentService : ITodoCommentService
     {
+        private readonly Workspace _workspace;
+        private KeepAliveSession _session;
```

Comment:
Why do you need this workspace?  Is it only for assertion purposes?  If so, can you document it as such.